packages/blade/src/components/TopNav/TabNav/TabNavItem.web.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import styled from 'styled-components';
+import type { TabNavItemProps } from './types';
+import BaseBox from '~components/Box/BaseBox';
+import getTextStyles from '~components/Typography/Text/getTextStyles';
+import { makeBorderSize, makeMotionTime, makeSpace } from '~utils';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import { makeAccessible } from '~utils/makeAccessible';
+
+const StyledTabNavItem = styled.a<{ $isActive?: boolean }>(({ theme, $isActive }) => {
+  return {
+    ...getTextStyles({
+      theme,
+      size: 'medium',
+      weight: 'medium',
+      color: 'interactive.text.gray.normal',
+    }),
+    flex: 1,
+    display: 'flex',
+    gap: makeSpace(theme.spacing[2]),
+    alignItems: 'center',
+    justifyContent: 'center',
+    height: '100%',
+    textDecoration: 'none',
+    whiteSpace: 'nowrap',
+    paddingTop: makeSpace(theme.spacing[3]),
+    paddingBottom: makeSpace(theme.spacing[3]),
+    paddingLeft: makeSpace(theme.spacing[4]),
+    paddingRight: makeSpace(theme.spacing[4]),
+    borderRadius: makeBorderSize(theme.border.radius.medium),
+    '&:hover': $isActive
+      ? {}
+      : {
+          backgroundColor: theme.colors.interactive.background.gray.default,
+        },
+  };
+});
+
+const StyledTabNavItemWrapper = styled(BaseBox)<{ isActive?: boolean }>(({ theme, isActive }) => {
+  const dividerHiderStyle = {
+    content: '""',
+    position: 'absolute',
+    top: '50%',
+    transform: 'translateY(-50%)',
+    width: 2,
+    height: '80%',
+    backgroundColor: theme.colors.surface.background.gray.subtle,
+  } as const;
+
+  return {
+    position: 'relative',
+    flexShrink: 0,
+    padding: `${makeSpace(theme.spacing[2])} ${makeSpace(theme.spacing[1])}`,
+    backgroundColor: isActive ? theme.colors.surface.background.gray.intense : 'transparent',
+    borderColor: isActive ? theme.colors.surface.border.gray.muted : 'transparent',
+    borderStyle: 'solid',
+    borderBottomWidth: 0,
+    borderWidth: makeBorderSize(theme.border.width.thin),
+    borderTopLeftRadius: makeBorderSize(theme.border.radius.medium),
+    borderTopRightRadius: makeBorderSize(theme.border.radius.medium),
+    // Animation
+    transform: isActive ? 'translateY(2px)' : 'none',
+    transition: `${makeMotionTime(theme.motion.duration.moderate)} ${
+      theme.motion.easing.standard.effective
+    }`,
+    transitionProperty: 'background, transform',
+
+    // Hide the left and right divider by overlaying them with a pseudo element as same color as the background
+    ...(isActive
+      ? {
+          ':before, :after': dividerHiderStyle,
+          ':before': {
+            left: -3,
+          },
+          ':after': {
+            right: -3,
```

Comment:
move these to variable maybe and explain where 3 comes from?

Reply:
Trial and error don't really have a solid logical explanation.